sim/aviron/src/lib/Cpu.zig

```diff
@@ -109,20 +101,92 @@ pub fn dump_system_state(cpu: *Cpu) void {
     std.debug.print("Y (r29:r28): 0x{X:0>4}\n", .{y_reg});
     std.debug.print("Z (r31:r30): 0x{X:0>4}\n", .{z_reg});
 
-    // Dump SRAM
-    std.debug.print("\nSRAM DUMP (0x0100-0x08FF, {} bytes):\n", .{cpu.sram.size});
-    for (0..cpu.sram.size) |i| {
-        if (i % 16 == 0) {
-            std.debug.print("0x{X:0>4}: ", .{0x0100 + i});
+    // Dump all registers
+    std.debug.print("\nREGISTERS:\n", .{});
+    for (cpu.regs, 0..) |reg, i| {
+        if (i % 8 == 0) std.debug.print("r{d:0>2}-r{d:0>2}: ", .{ i, @min(i + 7, 31) });
+        std.debug.print("{X:0>2} ", .{reg});
+        if ((i + 1) % 8 == 0) std.debug.print("\n", .{});
+    }
+
+    // Dump SRAM using absolute addresses based on configured base
+    const sram_base: u24 = cpu.sram.get_base();
+    const sram_end: u24 = sram_base + @as(u24, @intCast(cpu.sram.size - 1));
+    std.debug.print("\nSRAM DUMP (0x{X:0>4}-0x{X:0>4}, {d} bytes):\n", .{ sram_base, sram_end, cpu.sram.size });
+
+    const row_width = 16;
+    var prev_row: ?[row_width]u8 = null;
+    var prev_len: usize = 0;
+    var elided = false;
+
+    var i: usize = 0;
+    while (i < cpu.sram.size) : (i += @min(row_width, cpu.sram.size - i)) {
+        const row_len: usize = @min(row_width, cpu.sram.size - i);
+        var cur_row: [row_width]u8 = undefined;
+
+        var j: usize = 0;
+        while (j < row_len) : (j += 1) {
+            cur_row[j] = cpu.sram.read(sram_base + @as(u24, @intCast(i + j)));
```

Comment:
probably doesn't matter, but to avoid copying, we could maybe add a method that returns a slice of the underlying memory, which we can hold a handle to.